internal/files/repository.go

```diff
@@ -5,11 +5,33 @@
 package files
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"path/filepath"
+
+	"gopkg.in/yaml.v3"
 )
 
+func isGitWorktree(path string) error {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read %s: %w", path, err)
+	}
+	type gitWorktree struct {
+		GitDir string `yaml:"gitdir"`
+	}
+	var worktree gitWorktree
+	dec := yaml.NewDecoder(bytes.NewBuffer(content))
+	dec.KnownFields(true)
```

Comment:
Nit. Do we need this? This can be an issue if some other field is added to this file, right?

Reply:
TBH I don't know if that file could have other fields, so probably it's better to just remove it and ensure that at least the file contains "gitdir"...